example/readFromStream/readFromStream.cpp

```diff
@@ -0,0 +1,32 @@
+#include <fstream>
+#include "json/json.h"
+/*
+parse from stream,collect comments and capture error info.
+
+>g++ readFromStream.cpp -ljsoncpp -std=c++11 -o readFromStream
+>./readFromStream
+// comment head
+{
+    // comment before
+    "key" : "value"
+}
+// comment after
+// comment tail
+*/
+int main(int argc, char* argv[]) {
+  Json::Value jsonRoot;
+  Json::Value jsonItem;
+  std::ifstream ifs;
+  ifs.open(argv[1]);
+  jsonRoot.clear();
+
+  Json::CharReaderBuilder builder;
+  builder["collectComments"] = true;
+  JSONCPP_STRING errs;
+  if (!parseFromStream(builder, ifs, &jsonRoot, &errs)) {
+    // std::cout << errs <<std::endl;
+    return -1;
```

Comment:
-1 isn't a good main() exit code. I think it's out of range.
The EXIT_FAILURE would be best here.

Why not print the errs to std::cerr? Why comment it out?

Reply:
I 've tried this , but the EXIT_FAILURE and EXIT_SUCESS are redefined in sysytem's c++ lib.

Reply:
What do you mean "redefined"?
They're part of `<cstdlib>`.

Reply:
It shows that:
```c++
error: "EXIT_SUCCESS" redefined [-Werror]
#define EXIT_SUCCESS 1
In file included from /usr/include/c++/cstdlib:72:0,
                 from /usr/include/c++/5/ext/string_conversions.h:41,
                 from /usr/include/c++/5/bits/basic_string.h:5249,
                         ...
```

Reply:
OK, I got it, I should delete #define EXIT_SUCESS and EXIT_FAILURE. it's a system marco.